src/olcs/FeatureConverter.js

```diff
@@ -495,16 +504,55 @@ exports.prototype.olPolygonGeometryToCesium = function(layer, feature, olGeometr
       perPositionHeight: true
     });
 
-    outlineGeometry = new Cesium.PolygonOutlineGeometry({
-      // always update Cesium externs before adding a property
-      polygonHierarchy: hierarchy,
-      perPositionHeight: true
-    });
+    // Since Cesium doesn't yet support Polygon outlines on terrain yet (coming soon...?)
+    // we don't create an outline geometry if clamped, but instead do the polyline method
+    // for each ring. Most of this code should be removeable when Cesium adds
+    // support for Polygon outlines on terrain.
+    if (heightReference == Cesium.HeightReference.CLAMP_TO_GROUND) {
+      const width = this.extractLineWidthFromOlStyle(olStyle);
+      if (width > 0) {
+        outlinePrimitives = new Cesium.PrimitiveCollection();
+        const appearance = new Cesium.PolylineMaterialAppearance({
+          // always update Cesium externs before adding a property
+          material: this.olStyleToCesium(feature, olStyle, true)
+        });
+        const primitiveOptions = {
+          // always update Cesium externs before adding a property
+          appearance
+        };
+        let polylineGeometry = new Cesium.GroundPolylineGeometry({positions: hierarchy.positions, width});
+        primitiveOptions.geometryInstances = new Cesium.GeometryInstance({
+          geometry: polylineGeometry
+        }),
+        outlinePrimitives.add(new Cesium.GroundPolylinePrimitive(primitiveOptions));
+        if (hierarchy.holes) {
+          for (let i = 0; i < hierarchy.holes.length; ++i) {
+            polylineGeometry = new Cesium.GroundPolylineGeometry({positions: hierarchy.holes[i].positions, width});
+            primitiveOptions.geometryInstances = new Cesium.GeometryInstance({
+              geometry: polylineGeometry
+            });
+            outlinePrimitives.add(new Cesium.GroundPolylinePrimitive(primitiveOptions));
+          }
+        }
+      }
+    } else {
+      // Actually do the normal polygon thing. This should end the removable
+      // section of code described above.
+      outlineGeometry = new Cesium.PolygonOutlineGeometry({
+        // always update Cesium externs before adding a property
+        polygonHierarchy: hierarchy,
+        perPositionHeight: true
+      });
+    }
   }
 
   const primitives = this.wrapFillAndOutlineGeometries(
       layer, feature, olGeometry, fillGeometry, outlineGeometry, olStyle);
 
+  if (typeof outlinePrimitives !== 'undefined') {
```

Comment:
You declare the variable at the top of the function so you can simply do
```
if (outlinePrimitives) {...
```

Reply:
changed